core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -286,7 +286,10 @@ private boolean hasContentTypeOrCanAutoDetect(final RestRequest restRequest, fin
                     }
                 }
             } else if (restHandler != null && restHandler.supportsContentStream() && restRequest.header("Content-Type") != null) {
-                final String lowercaseMediaType = restRequest.header("Content-Type").toLowerCase(Locale.ROOT);
+                final String lowercaseContentType = restRequest.header("Content-Type").toLowerCase(Locale.ROOT);
+                final String[] elements = lowercaseContentType.split(";");
+                final String lowercaseMediaType = elements[0].trim();
+
```

Comment:
I feel we should still have some form of warning if you claim to be using a charset that isn't UTF-{8,16,32}, as per #22769. This might not be the time or the place to ask for that, however.